package_nso_to_oc/xe/xe_ospfv2.py

```diff
@@ -0,0 +1,584 @@
+
+def get_ospfv2_area(net_protocols, prot_index):
+    if (len(net_protocols) >= prot_index):
+        if not "openconfig-network-instance:ospfv2" in net_protocols[prot_index]: 
+            net_protocols[prot_index]["openconfig-network-instance:ospfv2"] = {}
+        if not "areas" in net_protocols[prot_index]["openconfig-network-instance:ospfv2"]:
+            net_protocols[prot_index]["openconfig-network-instance:ospfv2"]["areas"] = {"area": []}
+        
+        return net_protocols[prot_index]["openconfig-network-instance:ospfv2"]["areas"]["area"]
+    else:
+        # Sanity check, should not occur...
+        raise IndexError(f"The protocol index {prot_index} does not exist.")
+
+def get_area_by_id(ospfv2_area, area_id):
+    for area in ospfv2_area:
+        if area["identifier"] == area_id:
+            return area
+
+    new_area = {"identifier": area_id, "config": {"identifier": area_id}}
+    ospfv2_area.append(new_area)
+
+    return new_area
+
+def is_area_present_by_id(ospfv2_area, id):
+    for area in ospfv2_area:
+        if area.get("identifier", None) == id:
+            return True
+
+    return False
+
+def get_intf_by_intf_number(intf_attr, intf_number):
+    for intf in intf_attr:
+        if intf["name"] == intf_number:
+            return intf
+
+def process_ospf(net_protocols, vrf_interfaces, config_before, config_leftover, ospf_index, ospf):
+    ospf_leftover = config_leftover.get("tailf-ned-cisco-ios:router", {}).get("ospf")[ospf_index]
+    # If we got here, we init an empty dict and append to protocol list for future use.
+    net_protocols.append({})
+    prot_index = len(net_protocols) - 1
+    set_network_config(ospf_leftover, net_protocols, prot_index, ospf)
+    set_ospf2_global_config(ospf_leftover, net_protocols, prot_index, ospf)
+    set_graceful_restart_ietf(ospf_leftover, net_protocols, prot_index, ospf)
+    set_vrf_lite(ospf_leftover, net_protocols, prot_index, ospf)
+    set_default_info_originate(ospf_leftover, net_protocols, prot_index, ospf)
+    check_areas(ospf_leftover, net_protocols, vrf_interfaces, config_before, config_leftover, prot_index, ospf)
+    set_mpls_ldp_sync(ospf_leftover, net_protocols, prot_index, ospf)
+    set_timers_lsa(ospf_leftover, net_protocols, prot_index, ospf)
+    set_timers_spf(ospf_leftover, net_protocols, prot_index, ospf)
+
+def set_network_config(ospf_leftover, net_protocols, prot_index, ospf):
+    net_protocols[prot_index]["identifier"] = "OSPF"
+    net_protocols[prot_index]["name"] = f'{ospf.get("id")}'
+    temp_ospf = {"config": {
+        "identifier": "OSPF",
+        "name": f'{ospf.get("id")}',
+        "enabled": True
+    }}
+    net_protocols[prot_index].update(temp_ospf)
+
+    if ospf_leftover.get("id"):
+        del ospf_leftover["id"]
+
+def set_ospf2_global_config(ospf_leftover, net_protocols, prot_index, ospf):
+    if (not ospf.get("router-id") and not ospf.get("log-adjacency-changes") 
```

Comment:
We want these to be False if not in the configuration: log-adjacency-changes, compatible, suppression